pkg/controller/elasticsearch/certificates/transport/csr.go

```diff
@@ -113,6 +113,10 @@ func buildGeneralNames(
 }
 
 // buildCertificateCommonName returns the CN (and ES othername) entry for a given pod within a stack
```

Comment:
Let's use this PR as an opportunity to improve this comment?

```suggestion
// buildCertificateCommonName returns the CN (and ES otherName) entry for a given Elasticsearch Pod.
// If the user provided an otherName suffix in the spec, it prepends the pod name to it (<pod_name>.<user-suffix).
// Otherwise, it defaults to <pod_name>.node.<es_name>.es.local.
```